schema-layer/data-structures/vector.md

```diff
@@ -0,0 +1,361 @@
+
+Varying `width` always yields different root hashes regardless of data.
+
+## Structure
+
+### Parameters
+
+The only configurable parameter of an IPLD Vector is the `width`. This parameter must be consistent across all nodes in a Vector. Mutations cannot involve changes in `width` or joining multiple parts of a Vector with differing `width` values.
+
+`width` must be an integer, of at least `2`.
+
+### Node properties
+
+Each node in an IPLD vector stores the `width`, the `height` of the node, starting from `0` where values are stored, and a `data` array to contain values (for `height` `0`), or child node CIDs (for `height`s above `1`).
+
+### Schema
+
+```ipldsch
+type Vector struct {
+  width Int
+  height Int
+  data [ Value ]
+}
+
+type Value union {
+  | Link link
+  | Bool bool
+  | String string
+  | Bytes bytes
+  | Int int
+  | Float float
+  | Map map
+  | List list
+} representation kinded
+```
+
+**Constraints:**
+
+* Non-leaf (`height` greater than `0`) nodes only contain `Link`s to other `Vector` nodes in their `data` array.
+* `width` must be consistent across all nodes in a Vector.
+* `height` must be at least `0`.
+* `data` can contain between `0` and `width` elements.
+
+## Algorithm in detail
+
+### `Get(index)`
+
+`index` is a zero-based in all cases.
+
+`Get` can either be implemented as a recursive or iterative process.
+
+1. If `index` is less than zero, return undefined or an OOB indication.
+2. Calculate the maximum possible index for a tree with this `height` using `width`<sup>`height + 1`</sup>. If the index is greater than this maximum, returned undefined or an OOB indication.
+3. Calculate the local `dataIndex`, the index of this node's `data` array, using `floor(index / width`<sup>`height`</sup>`)`.
+4. If `dataIndex` is greater than or equal to the length of this node's `data` array, return undefined or an OOB indication.
+5. If `height` is `0`, return the element at the `dataIndex` position of this node's `data` array.
+6. If `height` is greater than `0`, retrieve the CID of the next node from the `dataIndex` position of this node's `data` array.
+7. Recurse or iterate from Step 1 with the next node with `index` set to `index % width`<sup>`height`</sup> to identify its place in the sub-tree.
+
+### `PushTail(value)`
+
+`PushTail` appends `value` to the tail of the data structure. This increases the Vector's `Size` by `1` and may cause zero or more overflows throughout the tree but will result in a single mutation at each `height` of the Vector, including the root and possibly the creation of a new root node at `height + 1` if an overflow cascades to the root.
+
+This algorithm can be implemented as a recursive or iterative process. The additional state needing to be transferred when mutating back up the tree from the mutated root may make recursive implementations awkward.
```

Comment:
you might be correct, I've soften this language to just:

> A recursive process will need to propagate additional state when mutating back up the tree from the mutated root.

I'll say more about why I'm leaning toward iterative processes below because I think you'll find it relevant.